inspectit-gepard-agent/src/test/java/rocks/inspectit/gepard/agent/integrationtest/spring/trace/TracingTest.java

```diff
@@ -52,69 +42,91 @@ void shouldNotSendSpansToBackendWhenNoScopesAreActive() throws Exception {
   private void assertSpans(
       Collection<ExportTraceServiceRequest> traces, String parentSpanName, String childSpanName) {
-    Stream<List<Span>> spanLists = getSpanLists(traces);
-
-    assertTrue(
-        spanLists.anyMatch(
-            spans -> {
-              Optional<Span> parentSpan = findSpan(spans, parentSpanName);
-              if (parentSpan.isEmpty()) return false;
-
-              Optional<Span> childSpan = findSpan(spans, childSpanName);
-              if (childSpan.isEmpty()) return false;
-
-              // We cannot compare the ByteStrings directly, because they are different objects
-              String childParentId = childSpan.get().getParentSpanId().toStringUtf8();
-              String parentId = parentSpan.get().getSpanId().toStringUtf8();
-              return childParentId.equals(parentId);
-            }));
+    List<Span> spans = getSpans(traces);
+
+    Optional<Span> parentSpan = findSpan(spans, parentSpanName);
+    Optional<Span> childSpan = findSpan(spans, childSpanName);
+    boolean spansExist = parentSpan.isPresent() && childSpan.isPresent();
+
+    assertTrue(spansExist);
+
+    // We cannot compare the ByteStrings directly, because they are different objects
+    String parentId = parentSpan.get().getSpanId().toStringUtf8();
+    String childParentId = childSpan.get().getParentSpanId().toStringUtf8();
+    String parentTraceId = parentSpan.get().getTraceId().toStringUtf8();
+    String childTraceId = childSpan.get().getTraceId().toStringUtf8();
+
+    assertEquals(parentId, childParentId);
+    assertEquals(parentTraceId, childTraceId);
   }
 
   /**
-   * This method asserts that no spans with the given names exist.
+   * This method asserts that no spans from inspectIT with the given names exist.
    *
    * @param traces the collection of traces
    * @param parentSpanName the name of the parent span
    * @param childSpanName the name of the child span
    */
   private void assertNoSpans(
       Collection<ExportTraceServiceRequest> traces, String parentSpanName, String childSpanName) {
-    Stream<List<Span>> spanLists = getSpanLists(traces);
+    List<Span> spans = getInspectItSpans(traces);
+
+    Optional<Span> parentSpan = findSpan(spans, parentSpanName);
+    Optional<Span> childSpan = findSpan(spans, childSpanName);
+    boolean noSpanExist = parentSpan.isEmpty() && childSpan.isEmpty();
 
-    assertTrue(
-        spanLists.anyMatch(
-            spans -> {
-              Optional<Span> parentSpan = findSpan(spans, parentSpanName);
-              Optional<Span> childSpan = findSpan(spans, childSpanName);
+    assertTrue(noSpanExist);
+  }
 
-              return parentSpan.isEmpty() && childSpan.isEmpty();
-            }));
+  /**
+   * Maps the provided traces to a collection of spans.
+   *
+   * @param traces the collection of traces
+   * @return the collection of span within the provided traces
+   */
+  private List<Span> getSpans(Collection<ExportTraceServiceRequest> traces) {
```

Comment:
Could you try to make this a little more readable?
For my taste, this is a little to nested.

Difference to getInspectIT Spans is only the filtering for INSTRUMENTATION_SCOPE_NAME in the end, right? 
So could you reuse the getSpans in getInspectitSpans, to avoid duplication?

Reply:
Yeah, the `ExportTraceServiceRequest` type is very nested as well. I would at least split it into two parts.